### PR TITLE
docs: linux: improve btrfs section

### PR DESCRIPTION
**Pull request by morbidrsa**
Improve the documentation and examples for btrfs.

Signed-off-by: Johannes Thumshirn <johannes.thumshirn@wdc.com>

### CODE REVIEW

**Comment by damien-lemoal** [comment]
There are many typos. Spell-check this please.

I think we also need the following:
* Requirements: libblkid version and btrfs progs version. For teach, add a link to the project and to the build/install instructions.
* An example with a ZNS drive

And I think we also need to mention at the beginning that the support is still "experimental" as there are open issues (link to the issues list).

**Comment by zdover23** [comment]
Damien,

I will vet this over the weekend. Leave the typos to me.

**Comment by damien-lemoal** [comment]
Thanks. But then better wait for the update first and review the revised version :)

**Comment by zdover23** [comment]
So be it.

**Comment by morbidrsa** [comment]
> There are many typos. Spell-check this please.

Really? I thought I did spell check it, will look again.
> 
> I think we also need the following:
> 
> * Requirements: libblkid version and btrfs progs version. For teach, add a link to the project and to the build/install instructions.

Sure will do.

> * An example with a ZNS drive

I have thought of this, but that wouldn't really change anything.

> And I think we also need to mention at the beginning that the support is still "experimental" as there are open issues (link to the issues list).

OK will do

File: docs/linux/fs.md [hunk 1]
```diff
@@ -579,23 +579,37 @@ must be specified.
 ## Btrfs
 
 *Btrfs* is a file system based on the copy-on-write (CoW) principle resulting in
-any block update to never be written in-place. Work is ongoing to add native ZBD
-support by changing the block allocation algorithm and block IO issuing code.
+any block update to never be written in-place. Support for zoned block
+devices in *btrfs* is present, but at the moment still in experimental.
```
Comment (zdover23) [inline docs/linux/fs.md:583]: "at the moment still in experimental" is ungrammatical, but I understand what is intended.

I suggest that this whole sentence ("Support... experimental.") be rewritten as follows:

*btrfs* currently supports zoned block devices, but that support is experimental.

File: docs/linux/fs.md [hunk 2]
```diff
@@ -610,16 +624,93 @@ The zones of a block group are reset to allow rewriting only when the block
...
+For *btrfs* volumes composed of multiple disks, restrictions are added to ensure
 that all disks have the same zone model and in the case of zoned block devices,
-the same zone size. This matches the existing *Btrfs* constraint that all device
+the same zone size. This matches the existing *btrfs* constraint that all device
 extents in a block group must have the same size.
 
-### Upstream Contribution
+All writes to data block groups use [Zone Append
+writing](../introduction/zns#zone-append), so that a high queue depth can be
+maintained without violating the device zone's sequential write constrains.
```
Comment (zdover23) [inline docs/linux/fs.md:634]: s/constrains/constraints/

**Comment by zdover23** [comment]
If I have anything to say about this, I say: Damien, merge this. I will give it an elegance rewrite. It seems to me to be of significant technical value, and should be incorporated.

If this is merged by the 23rd of December, I'll make sure that all the typos are correct and all the sentences are in lapidary technical English by Christmas.

File: docs/linux/fs.md (hunk 2 above)
Comment (morbidrsa) [reply docs/linux/fs.md:634]: fixed

File: docs/linux/fs.md (hunk 1 above)
Comment (morbidrsa) [reply docs/linux/fs.md:583]: done

**Comment by damien-lemoal** [comment]
Merged. Thanks !
I added one patch on top to mention the need for mq-deadline as the device block IO scheduler.

**Pull request merged**